### PR TITLE
Add Bind tests

### PR DESCRIPTION
**Pull request by vsomayaji**
## Overview

Adds tests for `Bind`, increasing coverage to 96.6%.

## Notes

```
--- PASS: TestBind (0.00s)
    --- PASS: TestBind/unsupported_parameter (0.00s)
    --- PASS: TestBind/error_getting_binding (0.00s)
    --- PASS: TestBind/existing_binding (0.00s)
    --- PASS: TestBind/conflicting_binding (0.00s)
    --- PASS: TestBind/error_getting_service (0.00s)
    --- PASS: TestBind/service_not_found (0.00s)
    --- PASS: TestBind/error_getting_instance (0.00s)
    --- PASS: TestBind/instance_not_found (0.00s)
    --- PASS: TestBind/error_describing_stack (0.00s)
    --- PASS: TestBind/error_getting_credentials (0.00s)
    --- PASS: TestBind/get_credentials (0.00s)
    --- PASS: TestBind/get_legacy_credentials (0.00s)
    --- PASS: TestBind/unsupported_scope (0.00s)
    --- PASS: TestBind/error_attaching_role_policy (0.00s)
    --- PASS: TestBind/attach_role_policy (0.00s)
```

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/broker/api_test.go [hunk 1]
```diff
@@ -320,34 +329,255 @@ func TestLastOperation(t *testing.T) {
...
 	}
-	reqContext := &broker.RequestContext{}
+}
 
-	expected := &broker.BindResponse{BindResponse: osb.BindResponse{Credentials: map[string]interface{}{}}}
-	actual, err := bl.Bind(bindReq, reqContext)
-	assertor.Equal(nil, err, "err should be nil")
-	assertor.Equal(expected, actual, "should succeed")
+func TestBind(t *testing.T) {
+	tests := []struct {
```
Comment (vsomayaji) [inline pkg/broker/api_test.go]: @jaymccon any objections to this style of writing test cases?

File: pkg/broker/types.go [hunk 2]
```diff
@@ -104,11 +105,11 @@ type DataStore interface {
 type GetAwsSession func(keyid string, secretkey string, region string, accountId string, profile string, params map[string]string) *session.Session
 
 type GetCfnClient func(sess *session.Session) CfnClient
-type GetSsmClient func(sess *session.Session) *ssm.SSM
+type GetSsmClient func(sess *session.Session) ssmiface.SSMAPI
```
Comment (vsomayaji) [inline pkg/broker/types.go]: I updated these to return the interface so that we can drop in mock implementations.

File: pkg/broker/util.go [hunk 3]
```diff
@@ -397,7 +397,7 @@ func getCredentials(service *osb.Service, outputs []*cloudformation.Output, ssmS
 		if err != nil {
 			return nil, err
 		} else if len(resp.InvalidParameters) > 0 {
-			return nil, fmt.Errorf("invalid parameters: %v", resp.InvalidParameters)
+			return nil, fmt.Errorf("invalid parameters: %v", aws.StringValueSlice(resp.InvalidParameters))
```
Comment (vsomayaji) [inline pkg/broker/util.go]: The tests help catch this 🐛.

File: pkg/broker/api_test.go (hunk 1 above)
Comment (jaymccon) [reply pkg/broker/api_test.go]: None. All the tests should look like this :)

**Pull request merged**